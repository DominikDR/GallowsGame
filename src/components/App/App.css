:global(body) {
    margin: 0;
}
.mainPage {
    min-height: 100vh;
    background-color: #232323;
    font-family: Arial, Helvetica, sans-serif;
}