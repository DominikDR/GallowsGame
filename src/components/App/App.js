import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchPhrase } from '../../fetchPhrase';
import { setGameState } from '../../actions/gameState';
import styles from './App.css';
import { Header } from '../Header';
import { Phrase } from '../Phrase';
import { SwitchKeyboard } from '../Keyboard/SwitchKeyboard';
import { ShowGallows } from '../ShowGallows';
import { GameOver } from '../GameOver';

export class App extends React.Component {
    async componentDidMount() {
        const { setGameState } = this.props;
        const data = await fetchPhrase();
        setGameState(data);
    }

    render() {
        return (
            <div className={styles.mainPage}>
                <Header />
                <Phrase />
                <ShowGallows />
                <SwitchKeyboard />
                <GameOver />
            </div>
        );
    }
}

const mapDispatchToProps = { setGameState };

App.propTypes = {
    setGameState: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(App);
