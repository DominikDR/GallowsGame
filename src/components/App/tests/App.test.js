import React from 'react';
import ReactDOM from 'react-dom';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { App } from '../App';

// 1 method mount- mounts the whole tree of components
// 2 shallow- do shallow render
// jest.fn() - creates anonymous function which always returns true

const initialState = {
    gameID: 0,
    category: '',
    encodedPhrase: '',
    failsCounter: 0,
    endState: null,
    letterStatus: {},
};
const mockStore = configureStore();
let component;
let store;

beforeEach(() => {
    store = mockStore(initialState);
    component = shallow(<App store={store} />);
});


describe('<App /> rendering', () => {
    it('renders the component', () => {
        expect(toJson(component)).toMatchSnapshot();
    });
});

/* 
describe('<App /> rendering', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
    });

    it('fetches data correctly', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('<Add /> lifecycle method invocations', () => {
    it('should change the state componentState componentDidMount method is invoked', () => {
        expect(component.state('componentState')).toEqual('mounted');
    });

    it('should call fetchPhrase during componentDidMount', () => {
        const fetchPhraseFake = jest.spyOn(App.prototype, 'fetchPhrase');
        expect(fetchPhraseFake).toHaveBeenCalledTimes(1);
    });

    it('should call setGameState during componentDidMount', () => {
        const setGameStateFake = jest.spyOn(App.prototype, 'setGameState');
        expect(setGameStateFake).toHaveBeenCalledTimes(1);
    });
});
 */