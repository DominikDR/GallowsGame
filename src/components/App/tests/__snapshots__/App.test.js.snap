// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> rendering renders the component 1`] = `
<div
  className="mainPage"
>
  <Header />
  <Connect(Phrase) />
  <Connect(ShowGallows) />
  <SwitchKeyboard />
  <Connect(GameOver) />
</div>
`;
